R/utils-classes.R

```diff
@@ -331,6 +331,26 @@ tail.bcdc_promise <- function(x, n = 6L, ...) {
 }
 
 
+#' @export
+names.bcdc_promise <- function(x) {
+  cols <- x[["cols_df"]]
+  query <- x[["query_list"]]
+
+  if (!is.null(query$propertyName)) {
+    select_cols <- strsplit(query$propertyName, ",")[[1]]
+    cols <- cols$col_name[cols$sticky | cols$col_name %in% select_cols]
+  } else {
+    cols <- cols$col_name
+  }
+
+  cols[cols == "SHAPE" | cols == "GEOMETRY"] <- "geometry"
+  geom_idx <- which(cols == "geometry")
+
+  cols[c((1:length(cols))[-geom_idx], geom_idx)]
```

Comment:
Small nit. I think this should be `seq_along(cols)` - so that if for some reason `length(cols) == 0` (unlikely I know) we're not trying to make a vector `1:0`
```suggestion
  cols[c(seq_along(cols)[-geom_idx], geom_idx)]
```